packages/wrangler/src/api/dev.ts

```diff
@@ -309,13 +242,20 @@ export function parseRequestInput(
 	input: RequestInfo = "/",
 	init?: RequestInit,
 	protocol: "http" | "https" = "http"
-): [RequestInfo, RequestInit | undefined] {
-	if (input instanceof Request) {
-		return [input, undefined];
-	}
-	const url = new URL(`${input}`, `${protocol}://${readyAddress}:${readyPort}`);
+): [RequestInfo, RequestInit] {
+	// Make sure URL is absolute
+	if (typeof input === "string") input = new URL(input, "http://placeholder");
+	// Adapted from Miniflare 3's `dispatchFetch()` function
+	const forward = new Request(input, init);
+	const url = new URL(forward.url);
+	forward.headers.set("MF-Original-URL", url.toString());
 	url.protocol = protocol;
 	url.hostname = readyAddress;
 	url.port = readyPort.toString();
-	return [url, init];
+	// Remove `Content-Length: 0` headers from requests when a body is set to
+	// avoid `RequestContentLengthMismatch` errors
+	if (forward.body !== null && forward.headers.get("Content-Length") === "0") {
+		forward.headers.delete("Content-Length");
+	}
```

Comment:
What returns the `Content-Length: 0` header? Some request clients? Why can't they be trusted to provide a good value here?